service/controller/v5/resource/encryptionkey/create.go

```diff
@@ -0,0 +1,60 @@
+package encryptionkey
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/rand"
+	"io"
+
+	"github.com/giantswarm/microerror"
+	apiv1 "k8s.io/api/core/v1"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-operator/service/controller/v5/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	var secret *apiv1.Secret
+	var encKey, encIV []byte
+
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	encKey = make([]byte, keySize)
+	if _, err := io.ReadFull(rand.Reader, encKey); err != nil {
+		return microerror.Mask(err)
+	}
+
+	encIV = make([]byte, aes.BlockSize)
+	if _, err := io.ReadFull(rand.Reader, encIV); err != nil {
+		return microerror.Mask(err)
+	}
+
+	secret = &apiv1.Secret{
+		Type: apiv1.SecretTypeOpaque,
+		ObjectMeta: apismetav1.ObjectMeta{
+			Name:      key.CertificateEncryptionName(customObject),
+			Namespace: key.CertificateEncryptionNamespace,
+			Labels: map[string]string{
+				key.LabelCluster: key.ClusterID(customObject),
```

Comment:
Adding the ManagedBy and Organization labels would be helpful IMO.

https://github.com/giantswarm/cluster-operator/blob/005a023b7ba2e0c040a4b08456c204df0f76faea/pkg/v11/resource/namespace/desired.go#L34-L36